proxy/logging/LogConfig.cc

```diff
@@ -765,14 +783,29 @@ LogConfig::update_space_used()
     if (sret != -1 && S_ISREG(sbuf.st_mode)) {
       total_space_used += (int64_t)sbuf.st_size;
 
-      if (auto_delete_rolled_files && LogFile::rolled_logfile(entry->d_name) && candidate_count < MAX_CANDIDATES) {
+      if (auto_delete_rolled_files && LogFile::rolled_logfile(entry->d_name) && total_candidate_count < MAX_CANDIDATES) {
         //
-        // then add this entry to the candidate list
+        // then check if the candidate belongs to any given log types with allowable space
         //
+        int len = strchr(strchr(entry->d_name, '.') + 1, '.') - entry->d_name;
```

Comment:
Why not use `TextView` here? This assumes the presence of '.' twice and will crash if that's not the case. Maybe
```
type_name = TextView(entry->d_name, strlen(entry->d_name));
type.name.remove_suffix(TextView(type_name).take_prefix_at('.').take_prefix_at('.')).size());
```
Alternatively, if it's really the suffix you want to remove, e.g. the name is "A.B.C" and  you want "A.B", then
```
TextView type_name(entry->d_name, strlen(entry->d_name));
type_name.take_suffix_at('.');
```